lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -0,0 +1,247 @@
+        if left == right
+          lranges << left
+        else
+          lranges << Range.new(left, right)
+        end
+      end
+      lranges.to_s.gsub('..', '-').delete('[').delete(']').delete(' ')
+    end
+
+    def cli_error_check(result)
+      # The NXOS bridge-domain cli does not raise an exception in some
+      # conditions and instead just displays a STDOUT error message;
+      # thus NXAPI does not detect the failure and we must catch it by
+      # inspecting the "body" hash entry returned by NXAPI. This
+      # bridge-domain cli behavior is unlikely to change.
+      fail result[2]['body'] if
+        result[2].is_a?(Hash) &&
+        /(ERROR:|Warning:)/.match(result[2]['body'].to_s)
+
+      # Some test environments get result[2] as a string instead of a hash
+      fail result[2] if
+        result[2].is_a?(String) &&
+        /(ERROR:|Warning:)/.match(result[2])
+    end
+
+    def create
+      result = config_set('bridge_domain', 'create', @bd_ids, @bd_ids)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def destroy
+      result = config_set('bridge_domain', 'destroy', @bd_ids, @bd_ids)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def self.allbds
+      hash = {}
+      bd_list = config_get('bridge_domain', 'all_bds')
+      return hash if bd_list.nil?
+
+      bd_list.each do |id|
+        hash[id] = BridgeDomain.new(id, false)
+      end
+      hash
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    # Bridge-Domain Shutdown case
+    def shutdown
+      result = config_get('bridge_domain', 'shutdown', @bd_ids)
+      # Valid result is either: "active"(aka no shutdown) or "shutdown"
+      result[/DOWN/] ? true : false
+    end
+
+    def shutdown=(val)
+      no_cmd = (val) ? '' : 'no'
+      result = config_set('bridge_domain', 'shutdown', @bd_ids, no_cmd)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_shutdown
+      config_get_default('bridge_domain', 'shutdown')
+    end
+
+    # Bridge-Domain name assigning case
+    def name
+      config_get('bridge_domain', 'name', @bd_ids)
+    end
+
+    def name=(str)
+      fail TypeError unless str.is_a?(String)
+      if str.empty?
+        result = config_set('bridge_domain', 'name', @bd_ids, 'no', '')
+      else
+        result = config_set('bridge_domain', 'name', @bd_ids, '', str)
+      end
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_name
+      sprintf('Bridge-Domain%s', @bd_ids)
+    end
```

Comment:
This is similar to the vlan name property, here even bridge-domains can be named.